### PR TITLE
Add a !datetime tag.

### PR DESCRIPTION
**Pull request by christinaholtNOAA**
**Synopsis**

Adds a `!datetime` tag and makes all tags lowercase.

Fixes user-identified gap identified in Discussion #595. 

**Type**

- [ ] Bug fix (corrects a known issue)
- [ ] Code maintenance (refactoring, etc. without behavior change)
- [ ] Documentation
- [x] Enhancement (adds new functionality)
- [ ] Tooling (CI, code-quality, packaging, revision-control, etc.)

**Impact**

- [ ] This is a breaking change (changes existing functionality)
- [x] This is a non-breaking change (existing functionality continues to work as expected)

**Checklist**

- [x] I have added myself and any co-authors to the PR's _Assignees_ list.
- [x] I have reviewed the documentation and have made any updates necessitated by this change.

### CODE REVIEW

File: docs/sections/user_guide/yaml/tags.rst [hunk 1]
```diff
@@ -23,6 +23,24 @@ Or explicit:
...
+   date1: 2024-09-01
+   date2: !datetime "{{ date1 }}"
+
+.. code-block:: text
+
+   % uw config realize -i ../input.yaml --output-format yaml
+   date1: 2024-09-01
+   date2: 2024-09-01 00:00:00
+
+The value provided to the tag must be in ISO format to be interpreted correctly by the ``!datetime`` tag.
```
Comment (maddenp-noaa) [inline docs/sections/user_guide/yaml/tags.rst:42]:
```suggestion
The value provided to the tag must be in ISO 8601 format to be interpreted correctly by the ``!datetime`` tag.
```
ISO is a vast set of standards, possibly including more than one date/time standard, so specificity might help. Could this maybe also be a hyperlink to https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat, which gives some examples and specifics on Python's implementation?

File: src/uwtools/tests/config/test_jinja2.py [hunk 2]
```diff
@@ -290,7 +291,14 @@ def test__deref_convert_no(caplog, tag):
     assert regex_logged(caplog, "Conversion failed")
 
 
-@mark.parametrize("converted,tag,value", [(3.14, "!float", "3.14"), (42, "!int", "42")])
+@mark.parametrize(
+    "converted,tag,value",
+    [
+        (datetime.fromisoformat("2024-09-09 00:00:00"), "!datetime", "2024-09-09 00:00:00"),
```
Comment (maddenp-noaa) [inline src/uwtools/tests/config/test_jinja2.py:297]:
```suggestion
        (datetime(2024, 9, 9, 0, 0), "!datetime", "2024-09-09 00:00:00"),
```

File: src/uwtools/tests/config/test_support.py [hunk 3]
```diff
@@ -87,6 +88,18 @@ def loader(self):
...
+    def test_datetime_no(self, loader):
+        ts = support.UWYAMLConvert(loader, yaml.ScalarNode(tag="!datetime", value="foo"))
+        with raises(ValueError):
+            ts.convert()
+
+    def test_datetime_ok(self, loader):
+        ts = support.UWYAMLConvert(
+            loader, yaml.ScalarNode(tag="!datetime", value="2024-08-09 12:22:42")
+        )
+        assert ts.convert() == datetime.fromisoformat("2024-08-09 12:22:42")
```
Comment (maddenp-noaa) [inline src/uwtools/tests/config/test_support.py:100]:
```suggestion
        assert ts.convert() == datetime(2024, 8, 9, 12, 22, 42)
```

**Review by maddenp-cu — APPROVED** [review]
Looks good, just a couple of minor suggestions.

**Review by nbharwani-cu — APPROVED** [review]

**Comment by christinaholtNOAA** [comment]
Oof! I did not realize I'd committed all the docs changes. I'm going to undo that and commit Paul's suggestions. Sorry about that!

**Review by WeirAE — APPROVED** [review]
Looks good to me with those little changes!

**Review by Byrnetp — APPROVED** [review]
LGTM!

**Review by elcarpenterNOAA — APPROVED** [review]

File: docs/sections/user_guide/yaml/tags.rst [hunk 4]
```diff
@@ -23,6 +23,24 @@ Or explicit:
...
+   date1: 2024-09-01
+   date2: !datetime "{{ date1 }}"
+
+.. code-block:: text
+
+   % uw config realize -i ../input.yaml --output-format yaml
+   date1: 2024-09-01
+   date2: 2024-09-01 00:00:00
+
+The value provided to the tag must be in ISO 8601 format to be interpreted correctly by the ``!datetime`` tag.
```
Comment (maddenp-noaa) [inline docs/sections/user_guide/yaml/tags.rst:42]: You might have decided against it for a good reason, but one more gentle plea for a link to https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat here, maybe like [ISO 8601 format](https://docs.python.org/3/library/datetime.html#datetime.datetime.fromisoformat), for examples of that and, especially, the notes in the Python docs about where its support differs from the standard.

Comment (christinaholtNOAA) [reply docs/sections/user_guide/yaml/tags.rst:42]: It's a great suggestion and I just let it slip through the cracks. I'll make the mod before I merge,  but that won't be until this afternoon.

**Pull request merged**